### ISSUE TITLE
How to debug: Child invisible and should not be added to tree.

### ISSUE THREAD

**Issue by pedia** [flutter/flutter#13808]
Sometimes I got this message:

> The following assertion was thrown during a scheduler callback:
> Child with id 73 is invisible and should not be added to the tree.
> 'package:flutter/src/semantics/semantics.dart': Failed assertion: line 671 pos 16:
> '!child.isInvisible'
> 
> 
How can I debug this error? Is there any way to dump all widgets from the tree?

[✓] Flutter (on Mac OS X 10.13.2 17C88, locale en-HK, channel master)
    • Flutter at /Users/odd/wa/flutter/flutter
    • Framework revision 235b64ed2f (2 weeks ago), 2017-12-14 20:38:39 -0800
    • Engine revision e0d19e2897
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.cf1de7d46cd88e204380e8f96a993439be56b24c

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.3)
    • Android SDK at /Users/odd/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-26, build-tools 26.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Community Edition (version 2017.3)
    • Flutter plugin version 20.0.3
    • Dart plugin version 173.3727.108

[✓] Connected devices
    • iPhone X • A786CB0E-0BD0-4698-919A-7440CFEDDD17 • ios • iOS 11.2 (simulator)

**Comment by Hixie** [comment]
cc @goderbauer

**Comment by goderbauer** [comment]
This looks like a duplicate of https://github.com/flutter/flutter/issues/13326, which has been fixed. You probably have to update to the latest version of flutter with `flutter upgrade` to get the fix.

Feel free to reopen, if you continue to have this problem after upgrading.
